src/fheroes2/gui/interface_list.h

```diff
@@ -417,15 +417,42 @@ namespace Interface
 
                 return true;
             }
-            if ( le.MousePressLeft( _scrollbar.getArea() ) && ( _size() > maxItems ) ) {
+            if ( le.MousePressLeft( _scrollbar.getArea() ) || le.MousePressLeft( rtAreaItems ) ) {
                 const fheroes2::Point mousePosition = le.GetMouseCursor();
 
                 const int32_t prevScrollbarX = _scrollbar.x();
                 const int32_t prevScrollbarY = _scrollbar.y();
 
                 UpdateScrollbarRange();
 
-                _scrollbar.moveToPos( mousePosition );
+                if ( le.MousePressLeft( _scrollbar.getArea() ) && ( _size() > maxItems ) ) {
+                    _scrollbar.moveToPos( mousePosition );
+                }
+
+                if ( le.MousePressLeft( rtAreaItems ) ) {
+                    if ( !le.isDragInProgress() ) {
+                        // Remember where has the drag started.
+                        _dragStartPos = mousePosition;
+                        le.RegisterDrag();
+
+                        // We have just started the drag, it might as well be a legitimate click.
+                        _lockClick = false;
+                    }
+
+                    const fheroes2::Point deltaPoint = _dragStartPos - mousePosition;
+                    const int delta = ( _scrollbar.isVertical() ? deltaPoint.y : deltaPoint.x );
```

Comment:
Since `deltaPoint` is used only here and only one part of it is used then we don't need to calculate both `x` and `y` at one time and calculate `delta` without creating the `deltaPoint`:
```suggestion
                    const int delta = _scrollbar.isVertical() ? ( _dragStartPos.y - mousePosition.y ) : ( _dragStartPos.x - mousePosition.x );
```